source/gui/__init__.py

```diff
@@ -585,10 +587,16 @@ def terminate():
 	import brailleViewer
 	brailleViewer.destroyBrailleViewer()
 
-	for instance, state in gui.SettingsDialog._instances.items():
-		if state is gui.SettingsDialog._DIALOG_DESTROYED_STATE:
+	# prevent race condition with object deletion
+	# prevent deletion of the object while we work on it.
+	_SettingsDialog = settingsDialogs.SettingsDialog
+	nonWeak: typing.Dict[_SettingsDialog, _SettingsDialog] = dict(_SettingsDialog._instances)
+
+	for instance, state in nonWeak.items():
+		if state is _SettingsDialog.DialogState.DESTROYED:
 			log.error(
-				"Destroyed but not deleted instance of settings dialog exists: {!r}".format(instance)
+				"Destroyed but not deleted instance of gui.SettingsDialog exists"
+				f": {instance.title} - {instance.__class__.__qualname__} - {instance}"

```

Comment:
Just a heads up that I've removed this code entirely in https://github.com/nvaccess/nvda/pull/12286

Reply:
I think that code is still useful, at least in theory. It warns us that we are holding onto a python object even though the corresponding wx.Window has already been destroyed. For some derivatives of SettingsDialog, there should only be one instance, so we keep track of the objects. In the past we have occasionally run into issues where objects are kept alive. This helps us to get an early warning during development. I'd be happy for it to be moved into a helper function with a comment that explains this though.